TestProject/output/BaseOutput.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace output;
+
+use options\Getopt;
+use cellularAutomat\Board;
+
+/**
+ * BaseOutput class which provides the functions for pluggable-output.
+ *
+ * @class BasOutput
+ */
+class BaseOutput
+{
+    /**
+     * @var int creates a new image*.png after every run.
+     */
+    protected $imageIndex = 0;
```

Comment:
This should be removed from the base-class. Not every output will need an imageIndex (for example console output) so the output should be responsible to define that variable if it needs one.